app/controllers/hosts_controller.rb

```diff
@@ -481,6 +481,19 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    find_multiple_ajax
+    res = {}
+    @hosts.includes(:compute_resource).each do |host|
+      if host.supports_power?
+        res[host.id] = host_running host
+      else
+        res[host.id] = "na"
+      end
+    end
+    render :json => res.to_json
```

Comment:
does it make sense to have a more formal json? e.g. {hostId: {status: on/off/unknown}}